### ISSUE TITLE
Get used time for current frame at given point.

### ISSUE THREAD

**Issue by HannulaTero** [YoYoGames/GameMaker-Bugs#8159]
### Is your feature request related to a problem?

Currently you can get time from start of the game with `current_time` or `get_timer()`, and also you can get time difference between current and previous frame using `delta_time`. But there is no way to get "used **frame-time** up until this point".

For example, when doing loading or calculations across the frames, it could be beneficial to know how much time has already been used for the game logic and rendering, and how much time is left of the frame budget. Let's say the game is 60FPS and total frame budget would be then around 16.7ms. Next let's assume we need to load assets, or generate landscape, or do other stuff which would generally take longer than the total frame budget. So the calculations must be split over the frames. How long these splits are can be hard percent of the total frame budget. But that poses a problem, as if game is already running and playable, game logic will be taking slice of this budget. Let's say 9ms, which would mean there is about 7.7ms left for other stuff.  For safety margin, let's say 7ms. The problem is how to get this number during the runtime. If you try do these calculations as last thing on the frame, you currently really can't directly know, how much time has already been spent. 

For demonstration, how get_frame_time() could be used:
```gml
// Repeat until free time has been used up.
// Assume these are done as last thing in the frame.
var _frameBudget = game_get_speed(gamespeed_microseconds) * 0.9;
do
{
  // Generate stuff or something. 
  // Do-until makes sure it will be executed at least once, 
  // but FPS might drop below target, if framebudget is already exceeded by gamelogic.
} until (get_frame_time() >= _frameBudget);

```

### Describe the solution you'd like

Similar to `get_timer()`, I would like to have `get_frame_time()` etc., which returns time used from beginning of the frame up until the point of function call.

### Describe alternatives you've considered

```gml
// Create persistent singleton object, and make sure it executes as very first thing in the frame.

// Instance create-event
depth = +15999;
time = 0;

// Instance begin-step event.
time = get_timer();

// For fetching currently used frametime.
function get_frame_time()
{
  return get_timer() - obj_frame_begin.time;
}
```

**Comment by gnysek** [comment]
It's not like it's totally not possible now, it's just not as accurate, as you're proposing in this Feature Request.

While lot of events might happen in frame, what we know is that "Begin Step" is one of first, and "Draw GUI End" is one of last.

Theoretically, by having only one object with "**Begin Step**" event (persistent one which is used by many games), you could just add `global.frame_start = get_timer();` on very first line, and then have:
```gml
// returns current frame time in microseconds (1/1_000_000 s)
function get_frame_time()
{
  return get_timer() - global.frame_start;
}

/// @desc returns frame usage in percent
function get_frame_usage() {
    static frame_time = game_get_speed(gamespeed_microseconds); //works only if game speed is not changed mid-game
    return get_frame_time() / frame_time;
}
```
And that's closest result we could get for now if I didn't missed anything.

Above solution have one big disadvantage - _Begin Step_ event would need to be reserved for only one instance in whole game, so having something more instance independent, which is surely performed before any code in a frame, would be still better - especially that it could already include engine time that passed in this frame for engine things.

**Comment by HannulaTero** [comment]
Indeed, having only one instance to use Begin step ensures this approach works properly. In my example I used high depth value to make it first to act, but that's "undefined behaviour" category.
